In [1]:
#imports
import pandas as pd
%matplotlib inline 
import matplotlib.pyplot as plt
from matplotlib import dates as mpl_dates
from matplotlib.pyplot import figure
import numpy as np
import seaborn as sns
from pandas.plotting import register_matplotlib_converters
register_matplotlib_converters()
import calendar

In [2]:
df_merchandise = pd.read_excel('Merchandise.xlsx', index_col="ProductID")

In [3]:
df_merchandise.head()

ProductType ProductColor ProductSize        ProductDescription  \
ProductID                                                                  
-2               None         None        None                      Lamp   
-1               None         None        None  Wrought Iron W. 5 Chairs   
 0             filler       filler      filler                    filler   
 1               None         None        None             Candle Holder   
 2               None         None        None                     Colum   

           OrigPrice   Price  EnterDate TransferDate  TransferPrice  \
ProductID                                                             
-2            147.60  147.60 2018-08-18   2018-08-18         147.60   
-1            124.75  124.75 2018-07-03   2018-07-03         124.75   
 0               NaN    0.00 1999-12-31   2018-04-14          68.00   
 1              2.00    2.00 2020-05-20   2020-05-20           2.00   
 2             49.50   49.50 2020-07-18   2020-07-18          49.50   

          TransferInd  ... ListOnline  CustNo  FixSplit  TaxRate  PhyDate  \
ProductID              ...                                                  
-2               Sold  ...      False     NaN     False      NaN      NaN   
-1               Sold  ...      False     NaN     False      NaN      NaN   
 0               Sold  ...      False     NaN     False      NaN      NaN   
 1               Sold  ...      False     NaN     False      NaN      NaN   
 2               Sold  ...      False     NaN     False      NaN      NaN   

          Auction2 ConCode BuyOut  CardFee  Reminded  
ProductID                                             
-2             NaN     NaN  False      0.0     False  
-1             NaN     NaN  False      0.0     False  
 0             NaN     NaN  False      0.0     False  
 1             NaN     NaN  False      0.0     False  
 2             NaN     NaN  False      0.0     False  

[5 rows x 50 columns]

In [4]:
df_merchandise.shape

(15308, 50)

In [5]:
df_merchandise.count()

ProductType           15308
ProductColor          15308
ProductSize           15308
ProductDescription    15308
OrigPrice             15307
Price                 15308
EnterDate             15308
TransferDate          10923
TransferPrice         15308
TransferInd           15308
TransferBy            15308
CustomerID            15308
ReqPrice              15308
InvoiceNumber         10906
Return                15308
MarkdownDate1          9976
MarkdownDate2          8497
MarkdownDate3          6976
LabelPrinted          15308
MercCut               15307
DownPymt              15308
DownWho                   0
DownDate                  0
ItemCount             15308
Date1                     0
Text1                   395
Num1                  15308
ExpireDate            15308
LayawayGrp                0
PCName                14925
DontPrintTag          15308
ItemNotes                 0
WholesalePrice        15308
BuyersFee             15308
DownPhone                 0
ResaleNow           

In [6]:
df_merchandise['DownPymt'].unique()
# will drop

array([0])

In [7]:
df_merchandise['Num1'].unique()
# will drop

array([0])

In [8]:
df_merchandise['Taxable'].unique()

array([nan, 'Y', 'N'], dtype=object)

In [9]:
df_merchandise['Taxable'].value_counts()
# will drop

Y    14910
N       15
Name: Taxable, dtype: int64

In [10]:
df_merchandise['NeverMD'].unique()

array([nan, 'Y', 'N'], dtype=object)

In [11]:
df_merchandise['NeverMD'].value_counts()
# will drop, unable to find documentation for this column description

N    13159
Y     1766
Name: NeverMD, dtype: int64

In [12]:
df_merchandise['TaxRate'].unique()
# will drop, these values will be worthless

array([nan,  0.])

In [13]:
df_merchandise['CardFee'].unique()
# will drop, these values will be worthless

array([ 0., nan])

In [14]:
# drop columns with no/many missing values and ones that are unnecessary
df_merchandise = df_merchandise.drop(['DownWho','DownDate','Date1','Text1','LayawayGrp','ItemNotes','Num1',
                                      'DownPhone','PreviousOwner','LabelPrinted','PCName','DontPrintTag',
                                      'Taxable','WholesalePrice','AuctionID','CustNo','PhyDate','Auction2',
                                      'DownPymt','ResaleNow','ListOnline','FixSplit','ConCode','NeverMD',
                                      'BuyOut','TaxRate','Reminded','CardFee'], axis=1)

In [15]:
# drop filler row
df_merchandise = df_merchandise.drop([0])
df_merchandise.head()

ProductType ProductColor ProductSize        ProductDescription  \
ProductID                                                                  
-2               None         None        None                      Lamp   
-1               None         None        None  Wrought Iron W. 5 Chairs   
 1               None         None        None             Candle Holder   
 2               None         None        None                     Colum   
 3               None         None        None             Credenza Lamp   

           OrigPrice   Price  EnterDate TransferDate  TransferPrice  \
ProductID                                                             
-2            147.60  147.60 2018-08-18   2018-08-18         147.60   
-1            124.75  124.75 2018-07-03   2018-07-03         124.75   
 1              2.00    2.00 2020-05-20   2020-05-20           2.00   
 2             49.50   49.50 2020-07-18   2020-07-18          49.50   
 3             71.10   71.10 2018-08-25   2018-08-25          71.10   

          TransferInd  ... ReqPrice  InvoiceNumber  Return  MarkdownDate1  \
ProductID              ...                                                  
-2               Sold  ...      0.0         8150.0   False            NaT   
-1               Sold  ...      0.0         7749.0   False            NaT   
 1               Sold  ...      0.0        13211.0   False            NaT   
 2               Sold  ...      0.0        13842.0   False            NaT   
 3               Sold  ...      0.0         8225.0   False            NaT   

           MarkdownDate2 MarkdownDate3 MercCut ItemCount  ExpireDate  \
ProductID                                                              
-2                   NaT           NaT     0.0         0  2018-11-16   
-1                   NaT           NaT     0.5         0  2018-10-01   
 1                   NaT           NaT     0.0         0  2020-08-18   
 2                   NaT           NaT     0.5         0  2020-10-16   
 3                   NaT           NaT     0.0         0  2018-11-23   

           BuyersFee  
ProductID             
-2               0.0  
-1               0.0  
 1               0.0  
 2               0.0  
 3               0.0  

[5 rows x 22 columns]

In [16]:
df_merchandise.dtypes

ProductType                   object
ProductColor                  object
ProductSize                   object
ProductDescription            object
OrigPrice                    float64
Price                        float64
EnterDate             datetime64[ns]
TransferDate          datetime64[ns]
TransferPrice                float64
TransferInd                   object
TransferBy                    object
CustomerID                     int64
ReqPrice                     float64
InvoiceNumber                float64
Return                          bool
MarkdownDate1         datetime64[ns]
MarkdownDate2         datetime64[ns]
MarkdownDate3         datetime64[ns]
MercCut                      float64
ItemCount                      int64
ExpireDate            datetime64[ns]
BuyersFee                    float64
dtype: object

In [17]:
df_merchandise['ProductType'].value_counts()

Accessory          5266
Chair              1194
Decor               655
Table               402
None                384
Pillow              371
Art                 365
Sofa                296
Rug                 293
End Table           228
Lamp                224
Picture             221
Desk                171
Light               155
Bed                 140
Barstool            124
Mirror              113
Coffee Table        111
Dining Table        104
Chest               104
Console              99
Ottoman              93
Cabinet              90
Book                 86
Sidetable            79
Chandelier           77
Bench                65
Nightstand           63
Dresser              60
Clock                54
                   ... 
Gold Leaf Box         1
Pot Rack              1
Navy/white Stri       1
Entry Stand           1
Lightspace            1
Antigue               1
Wall Sconse           1
Orange Vase           1
Rug Green             1
OK Made               1
Drapery Panel   

In [18]:
products = df_merchandise['ProductType'].value_counts()
products[products < 10].shape
# there are 1103 entries where the total count for a given Product Type appears less than 10 times. 
# I will likely exclude these entries from the dataframe when analyzing products

(1103,)

In [19]:
df_merchandise['ProductColor'].value_counts()

None        14779
White          53
Brown          44
Black          29
Ivory          25
Flint          22
Driftwoo       19
Silver         17
Linen          13
Sapphire       12
Wheat          12
Seaglass       12
Gold           11
Blue           11
Red            10
Teal            9
Gray            9
Wood            9
Navy            8
Multi           8
Oatmeal         8
Metal           7
Misc            7
Clear           6
Tommy           6
Green           5
Yellow          5
Small           4
Bronze          4
Cream           3
            ...  
Yosimite        1
Metalic         1
Wooden          1
Verde           1
2 Leaves        1
Brn/Gold        1
Bernhard        1
Cherry          1
Tan             1
King            1
175.00          1
5.              1
Gld Red         1
Gry/oran        1
1000.           1
Marble T        1
Drk Gree        1
250.00          1
Hand Wov        1
Gld Oran        1
Camel           1
Red/gold        1
Beige Pr        1
Iron            1
Beige/ta  

In [20]:
products_color = df_merchandise['ProductColor'].value_counts()
products_color[products_color != 'None'].shape

/Users/zachcornelison/anaconda3/lib/python3.7/site-packages/pandas/core/ops.py:1649: FutureWarning: elementwise comparison failed; returning scalar instead, but in the future will perform elementwise comparison
  result = method(y)


(134,)

Since there's only 134 entires where the Product Color is any description other than "None", I will remove this column from the dataset as it will not be useful for analysis. 

In [21]:
df_merchandise = df_merchandise.drop(['ProductColor'], axis=1)

In [22]:
df_merchandise['ProductSize'].value_counts()

None          14798
Jubilee          94
Park Avenu       39
Small            18
Queen            18
Medium           17
King             14
Large            13
Pomeroy          11
53006225         10
B446              8
CM3500            8
CM1700            8
CM4550            8
CM3200            7
53005996          7
Wood              6
9ft X 4ft         6
22x22             5
Round             3
53005413el        3
98 Inch           2
Clkb2a175         2
Xpgd10307-        2
V950434           2
Chair             2
8x10              2
53003675el        2
53050040          2
ARTAC0436C        2
              ...  
CM5268            1
53W 14.5L         1
CLKA9B364N        1
51030245          1
Full              1
24x30             1
CM5273            1
CM3200Q/F         1
51003645el        1
FULL              1
ARTAE2028         1
Cm1700            1
CL19601137        1
9 Oz.             1
100x17x43         1
51003618el        1
48                1
ARTAE1920D        1
ARTAE0058         1


In [23]:
products_size = df_merchandise['ProductSize'].value_counts()
products_size[products_size != 'None'].shape

(201,)

I will again drop this column as only 201 of the rows contain useful information regarding Product Size

In [24]:
df_merchandise = df_merchandise.drop(['ProductSize'], axis=1)

In [25]:
df_merchandise['ProductDescription'].value_counts()

Pillow                       611
Lamp                         436
Picture                      347
Art                          315
Chair                        296
Rug                          228
Mirror                       216
Vase                         203
Decor                        202
Chandelier                   145
Barstool                     108
Free                          98
Sofa                          95
Book                          88
Desk                          87
Nightstand                    78
Sidetable                     69
Plate                         68
Office Chair                  65
Finial                        65
Coffeetable                   61
Table                         59
Frame                         56
Pewter Figurine               53
Ottoman                       53
Bookcase                      52
Light Fixture                 51
Painting                      50
Console                       49
Bowl                          48
          

want to add product descriptions to product type field if product type field has "none" for the entry and the descrtipion has something relevant to replace with (i.e. sofa, chair, table, rug, etc.).